lib/carmen/region_collection.rb

```diff
@@ -16,6 +16,17 @@ module Carmen
   class RegionCollection < Array
     include Querying
 
+    # Filters the regions in this collection by type.
+    #
+    # type - The String type to filter by
+    #
+    # Returns a region collection containing all the regions with the supplied
+    # type.
+    def typed(type)
+      results = select{ |r| r.type.downcase == type.downcase }
```

Comment:
I think all ISO source regions are already downcase. Otherwise maybe we should downcase the type at load to avoid doing it every time you search by type. 

Same thing with `type.downcase`, I think it should be done outside the select block to avoid doing it repetitively.

```ruby
test = Array.new(10000) { rand.to_s }

puts Benchmark.measure { test.select { |x| x.downcase == "ABCD".downcase }  }
0.000000   0.000000   0.000000 (  0.003354)

downcased = "abcd"
puts Benchmark.measure { test.select { |x| x == downcased }  }
0.000000   0.000000   0.000000 (  0.000718)
```